lib/github-api.ts

```diff
@@ -268,23 +300,24 @@ const githubGraphqlRequest = async <T>(query: string, variables: object) => {
 // }
 // 2. Query contains a variable called endCursor that is used to page through results.
 // See: https://docs.github.com/en/graphql/guides/using-pagination-in-the-graphql-api to learn more about these assumptions.
-async function githubGraphqlRequestPaging<RESPONSE, RETURN>(
+async function githubGraphqlRequestPaging<RESPONSE>(
   query: string,
-  variables: { [key: string]: string },
-  processResponse: (data: RESPONSE) => RETURN | null,
-): Promise<RETURN | null> {
-  let returnValue: RETURN | null = null;
-
+  variables: { [key: string]: string | number },
+  processResponse: (data: RESPONSE) => Promise<boolean>,
+): Promise<void> {
   // deno-lint-ignore no-explicit-any
-  function findPageInfo(obj: any): { hasNextPage: boolean; endCursor: string } {
+  function findPageInfo(_obj: any): { hasNextPage: boolean; endCursor: string } {
+    // Create a shallow copy of the object to avoid modifying the original
+    const obj = { ..._obj };
+
+    // nodes is the JSON response. It could be a really big object. Do not perform recursion on it, so let's delete it. 
+    delete obj["nodes"];
```

Comment:
Encountered a bug where recursion was getting a stackoverflow. Fixed by ignoring the large nodes object.